qemu/tests/win_driver_install_by_installer.py

```diff
@@ -0,0 +1,330 @@
+    install_path = params["install_path"]
+    check_installed_cmd = params["check_installed_cmd"] % install_path
+    installer_pkg_check_cmd = params["installer_pkg_check_cmd"]
+
+    # qemu version check test config
+    qemu_ga_pkg = params["qemu_ga_pkg"]
+    gagent_pkg_info_cmd = params["gagent_pkg_info_cmd"]
+    gagent_uninstall_cmd = params["gagent_uninstall_cmd"]
+
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+
+    devcon_path = utils_misc.set_winutils_letter(session,
+                                                 params["devcon_path"])
+    status, output = session.cmd_status_output("dir %s" % devcon_path,
+                                               timeout=OPERATION_TIMEOUT)
+    if status:
+        test.error("Not found devcon.exe, details: %s" % output)
+
+    if params.get("need_uninstall", "no") == "yes":
+
+        driver_name_list = ['netkvm', 'viorng', 'vioser',
+                            'balloon', 'pvpanic', 'vioinput',
+                            'viofs', 'viostor', 'vioscsi']
+
+        device_hwid_list = ['"PCI\\VEN_1AF4&DEV_1000" "PCI\\VEN_1AF4&DEV_1041"',
+                            '"PCI\\VEN_1AF4&DEV_1005" "PCI\\VEN_1AF4&DEV_1044"',
+                            '"PCI\\VEN_1AF4&DEV_1003" "PCI\\VEN_1AF4&DEV_1043"',
+                            '"PCI\\VEN_1AF4&DEV_1002" "PCI\\VEN_1AF4&DEV_1045"',
+                            '"ACPI\\QEMU0001"', '"PCI\\VEN_1AF4&DEV_1052"',
+                            '"PCI\\VEN_1AF4&DEV_105A"',
+                            '"PCI\\VEN_1AF4&DEV_1001" "PCI\\VEN_1AF4&DEV_1042"',
+                            '"PCI\\VEN_1AF4&DEV_1004" "PCI\\VEN_1AF4&DEV_1048"']
+
+        device_name_list = ["Red Hat VirtIO Ethernet Adapter", "VirtIO RNG Device",
+                            "VirtIO Serial Driver", "VirtIO Balloon Driver",
+                            "QEMU PVPanic Device", "VirtIO Input Driver",
+                            "VirtIO FS Device", "Red Hat VirtIO SCSI controller",
+                            "Red Hat VirtIO SCSI pass-through controller"]
+        for driver_name, device_name, device_hwid in zip(driver_name_list,
+                                                         device_name_list, device_hwid_list):
+            uninstall_driver(session, driver_name, device_name, device_hwid)
+
+        error_context.context("Install 'qemu-guest-agent' package in guest.",
+                              logging.info)
+        vol_virtio_key = "VolumeName like '%virtio-win%'"
+        vol_virtio = utils_misc.get_win_disk_vol(session, vol_virtio_key)
+        qemu_ga_pkg_path = r"%s:\%s\%s" % (vol_virtio, "guest-agent", qemu_ga_pkg)
+        gagent_install_cmd = params["gagent_install_cmd"] % qemu_ga_pkg_path
+        s_inst, o_inst = session.cmd_status_output(gagent_install_cmd)
+        if s_inst != 0:
+            test.fail("qemu-guest-agent install failed,"
+                      " the detailed info:\n%s." % o_inst)
```

Comment:
test.error is more suitable.

Reply:
for qemu-guest-agent install and uninstall not the main steps for this test case, so test.error is more suitable? if my understanding is right?

Reply:
Yes, that's true